src/applications/personalization/dashboard/tests/actions/claims.unit.spec.jsx

```diff
@@ -0,0 +1,164 @@
+import { expect } from 'chai';
+import sinon from 'sinon';
+
+import {
+  mockFetch,
+  setFetchJSONFailure,
+  setFetchJSONResponse,
+} from '@department-of-veterans-affairs/platform-testing/helpers';
+
+import { createClaimsSuccess } from '../../mocks/claims';
+import { createAppealsSuccess } from '../../mocks/appeals-success';
+import { getAppealsV2, getClaims } from '../../actions/claims';
+
+import {
+  FETCH_CLAIMS_PENDING,
+  FETCH_CLAIMS_ERROR,
+  FETCH_CLAIMS_SUCCESS,
+} from '../../utils/claims-helpers';
+import {
+  FETCH_APPEALS_PENDING,
+  FETCH_APPEALS_SUCCESS,
+  USER_FORBIDDEN_ERROR,
+  RECORD_NOT_FOUND_ERROR,
+  VALIDATION_ERROR,
+  BACKEND_SERVICE_ERROR,
+  FETCH_APPEALS_ERROR,
+} from '../../utils/appeals-helpers';
+
+describe('/actions/claims', () => {
+  describe('getAppealsV2', () => {
+    let dispatchSpy;
+    let oldDataLayer;
+    beforeEach(() => {
+      mockFetch();
+      oldDataLayer = global.window.dataLayer;
+      global.window.dataLayer = [];
+      dispatchSpy = sinon.spy();
+    });
+    afterEach(() => {
+      global.window.dataLayer = oldDataLayer;
+    });
+
+    it('should dispatch FETCH_APPEALS_PENDING action', () => {
+      getAppealsV2()(dispatchSpy);
+      expect(dispatchSpy.firstCall.args[0].type).to.equal(
+        FETCH_APPEALS_PENDING,
+      );
+    });
+
+    it('should dispatch a FETCH_APPEALS_SUCCESS action', () => {
+      setFetchJSONResponse(global.fetch.onCall(0), createAppealsSuccess());
+      const thunk = getAppealsV2();
+
+      const dispatch = action => {
+        dispatchSpy(action);
+        if (dispatchSpy.callCount === 2) {
+          expect(dispatchSpy.secondCall.args[0].type).to.equal(
+            FETCH_APPEALS_SUCCESS,
+          );
+        }
+      };
+      thunk(dispatch);
+    });
+
+    const appealsErrors = {
+      403: USER_FORBIDDEN_ERROR,
+      404: RECORD_NOT_FOUND_ERROR,
+      422: VALIDATION_ERROR,
+      502: BACKEND_SERVICE_ERROR,
+      504: FETCH_APPEALS_ERROR, // works for any unspecified error code
+    };
+
+    Object.keys(appealsErrors).forEach(code => {
+      it(`Dispatches ${
+        appealsErrors[code]
+      } when GET fails with ${code}`, done => {
+        setFetchJSONResponse(
+          global.fetch.onCall(0),
+          Promise.reject({
+            errors: [{ status: `${code}` }],
+          }),
+        );
+        const thunk = getAppealsV2();
+        const dispatch = sinon.spy();
+        thunk(dispatch)
+          .then(() => {
+            const action = dispatch.secondCall.args[0];
+            expect(action.type).to.equal(appealsErrors[code]);
+          })
+          .then(done, done);
+      });
+    });
+  });
+
+  describe('getClaims', () => {
+    let dispatchSpy;
+    let oldDataLayer;
+    beforeEach(() => {
+      mockFetch();
+      oldDataLayer = global.window.dataLayer;
+      global.window.dataLayer = [];
```

Comment:
I had one more thought about the `dataLayer` stuff. This feels like a code smell to me in that we aren't testing the dataLayer array logic or anything that relies on it, so it probably shouldn't be something we assert on.

A quick grep of the repo shows this happening around 143 times in 48 files, so I think it's fine to leave the way it is, but we could probably use dependency injection here to remove the `recordEvent` implementation reliance.

Reply:
yeah.. in all transparency i _did_ grab this from another test file. it does feel clunky and there's probably a better way to test this, possibly something like this:

https://github.com/department-of-veterans-affairs/vets-website/blob/4149157463247878102feff50c1b78f88c2c1d90/src/applications/personalization/profile/tests/util/direct-deposit.unit.spec.js#L84C1-L100C8

```
    it('records lighthouse analytics event', () => {
      const errorMessage = 'test-error-message';
      client.recordCNPEvent({
        status: API_STATUS.FAILED,
        method: 'PUT',
        extraProperties: {
          'error-key': errorMessage,
        },
      });


      expect(recordEventSpy.firstCall.args[0]).to.deep.equal({
        event: 'api_call',
        'api-name': 'PUT /profile/direct_deposits/disability_compensations',
        'api-status': API_STATUS.FAILED,
        'error-key': errorMessage,
      });
    });
```

Reply:
i think i want to work on improvements in another ticket.. i just want to get our increased coverage out asap!

Reply:
I think these sorts of future improvements to stub and mock out global singletons and things can go a long way helping with coverage, along with speed and reliability :). Might be some more significant refactoring to do for those tho, so follow up ticket(s) makes sense to me 👍